core/remote/selectionmodelserver.cpp

```diff
@@ -29,16 +29,37 @@
 #include "selectionmodelserver.h"
 #include "server.h"
 
+#include <QTimer>
+
 using namespace GammaRay;
 
 SelectionModelServer::SelectionModelServer(const QString& objectName, QAbstractItemModel* model, QObject* parent):
-  NetworkSelectionModel(objectName, model, parent)
+  NetworkSelectionModel(objectName, model, parent), m_timer(new QTimer(this))
 {
+  m_timer->setSingleShot(true);
+  m_timer->setInterval(125);
+  Q_ASSERT(model);
+  // We do use a timer to group requests to avoid network overhead
+  connect(model, SIGNAL(rowsInserted(QModelIndex,int,int)), m_timer, SLOT(start()));
+  connect(model, SIGNAL(columnsInserted(QModelIndex,int,int)), m_timer, SLOT(start()));
+  connect(model, SIGNAL(rowsMoved(QModelIndex,int,int,QModelIndex,int)), m_timer, SLOT(start()));
+  connect(model, SIGNAL(rowsRemoved(QModelIndex,int,int)), m_timer, SLOT(start()));
+  connect(model, SIGNAL(columnsRemoved(QModelIndex,int,int)), m_timer, SLOT(start()));
+  connect(model, SIGNAL(columnsMoved(QModelIndex,int,int,QModelIndex,int)), m_timer, SLOT(start()));
```

Comment:
Ah, so Qt4 compatibility is why we can't use the virtuals in QISM? I was wondering already, that should avoid the flag issue. Ok, then we probably don't have a choice here.